js/parts/Series.js

```diff
@@ -3112,10 +3114,10 @@ H.Series = H.seriesType('line', null, { // base series options
                     // are not touched.
                     oldData[pointIndex].touched = true;
 
-                    // Speed optimize by only searching from last known index.
+                    // Speed optimize by only searching after last known index.
                     // Performs ~20% bettor on large data sets.
                     if (requireSorting) {
-                        lastIndex = pointIndex;
+                        lastIndex = pointIndex + 1;
```

Comment:
@TorsteinHonsi 
This should resolve the problem for sorted data - checking from next point to avoid using the same point again.